### ISSUE TITLE
Channel apparently successfully attaches, but OnAttachTimeout fires anyway

### ISSUE THREAD

**Issue by SimonWoolf** [ably/ably-dotnet#205]
Log from a customer (extracted from https://support.ably.io/helpdesk/tickets/2939 ):

```
IRealTimeClient.Channels.Get:

11:37:43.958 Setting up timer '#testchnl:Attached awaiter timer' to run action after 10 seconds.
11:37:43.959 #testchnl: Changing state to: 'Attaching'. 
11:37:43.960 RealtimeChannel.SendMessage:Attach
11:37:43.960 Current state: Connected. Sending message: { action=Attach }
11:37:43.962 Sending message (Attach) to transport

[Callback]Attaching: 

11:37:45.518 Proccessing queued message: { action=Attached }
11:37:45.518 #testchnl: Changing state to: 'Attached'. 
11:37:45.520 StartSync | Channel: testchnl, SyncInProgress: False

[Callback]Attached: 

11:37:50.005 Proccessing queued message: { action=Heartbeat }
11:37:53.960 Timer '#testchnl:Attached awaiter timer' interval 10 seconds elapsed and calling action.
11:37:53.963 #testchnl didn't Attach within 00:00:10. Setting state back to Attaching
11:37:53.964 #testchnl: Changing state to: 'Attaching'. Error: Reason: Channel didn't attach within the default timeout; Code: 50000
11:37:53.964 RealtimeChannel.SendMessage:Attach
11:37:53.964 Current state: Connected. Sending message: { action=Attach }
11:37:53.964 Sending message (Attach) to transport
11:37:53.972 Proccessing queued message: { action=Attached }
11:37:53.972 #testchnl: Changing state to: 'Attached'. 
11:37:53.972 StartSync | Channel: testchnl, SyncInProgress: True

[Callback]Attaching: Reason: Channel didn't attach within the default timeout; Code: 50000

[Callback]Attached:
```

**Comment by SimonWoolf** [comment]
More information from the customer:
```
I am using v0.8.9 for the client lib in Unity.

I am also using an alternative websocket transport as shown in this repo.
https://github.com/ably-forks/ably-dotnet-alternative-transports

This is the websocket transport I am using.
https://github.com/sta/websocket-sharp
```

**Comment by withakay** [comment]
Unity is not a supported platform (yet) but if we can can get some code to reproduce this we can take a look. Ideally a sample Unity project that demonstrates the issue, if possible?

**Comment by yunxuchen** [comment]
Hi, I have uploaded a repository with the sample project. I am using Unity 2017.4.0f1 with .Net 4.6 runtime version.
https://github.com/yunxuchen/ably_test

**Comment by DavidInBremen** [comment]
Hello

I'm also seeing this issue; it's not related to Unity and it sometimes causes problems with moving from Detached, back to Detaching and then trying to Detach again, which can result in an error that the channel doesn't exist anymore. I've attached a simple console app to demonstrate.

I thought it may be related to the number of channels, as I initially saw it with 3 channels, but it happens with just a single channel.

[Ably.txt](https://github.com/ably/ably-dotnet/files/1928123/Ably.txt)

Kind regards
David

**Comment by withakay** [comment]
Thanks for the examples @yunxuchen and @DavidInBremen.
DavidInBremen I ran your example and can confirm that this is a bug. 
I will bump it up the priority list.

**Comment by withakay** [comment]
Fixed in forth coming 0.8.11 release
